packages/dds/map/src/map.ts

```diff
@@ -330,18 +330,17 @@ export class SharedMap extends SharedObject<ISharedMapEvents> implements IShared
     * {@inheritDoc @fluidframework/shared-object-base#SharedObject.loadCore}
     */
     protected async loadCore(storage: IChannelStorageService) {
-        const header = await storage.read(snapshotFileName);
-
-        const data = fromBase64ToUtf8(header);
+        const blob = await storage.readBlob(snapshotFileName);
+        const header = blobToString(blob);
         // eslint-disable-next-line @typescript-eslint/ban-types
-        const json = JSON.parse(data) as object;
+        const json = JSON.parse(header) as object;
         const newFormat = json as IMapSerializationFormat;
         if (Array.isArray(newFormat.blobs)) {
             this.kernel.populateFromSerializable(newFormat.content);
             await Promise.all(newFormat.blobs.map(async (value) => {
-                const blob = await storage.read(value);
-                const blobData = fromBase64ToUtf8(blob);
-                this.kernel.populateFromSerializable(JSON.parse(blobData) as IMapDataObjectSerializable);
+                const newBlob = await storage.readBlob(value);
```

Comment:
Sequence of readBlob, blobToString & JSON.parse is repeated in different places. Ocasionally 3rd step is serializer.parse().
I wonder if we want to move deserializeBlob() to more shared place, make it a bit more generic (make serializer = undefined as default argument) and call JSON.parse if serializer === undefined.
Then code like here can be replaced to just calling deserializeBlob

Reply:
I'm generally against adding too many helper functions here to save a line or two.  I feel masking the logic in a helper function like this is harder to read and understand than just the explicit steps, particularly if the helper has special behavior on serializer === undefined.

With our design choice to move the storage layer to only expose binary blobs, I think it's appropriate and consistent for any consumer of the storage layer to acknowledge this format and perform their own format conversion as they need (including custom JSON parsing, etc. if applicable).  Otherwise we might as well just keep readString on the storage layer.

Reply:
We can keep it as is - readAndParse & deserializeBlob (i.e. two functions, not one), but I do not see good reason to start inlining functions like that. Taken to extreme, that would result in no code reuse, and functions 10K lines long (common practice of open/load code in Office in 90s).